src/snarky.d.ts

```diff
@@ -18,31 +18,124 @@ export {
 export { Snarky, Test, JsonGate };
 
 /**
- * `Provable<T>` is the general circuit type interface. Provable interface describes how a type `T` is made up of field elements and auxiliary (non-field element) data.
+ * `Provable<T>` is the general circuit type interface in SnarkyJS. `Provable<T>` interface describes how a type `T` is made up of {@link Field} elements and "auxiliary" (non-provable) data.
  *
- * You will find `Provable<T>` as the required input type in a few places in SnarkyJS. One convenient way to create a `Provable<T>` is using `Struct`.
+ * `Provable<T>` is the required input type in a few places in SnarkyJS. One convenient way to create a `Provable<T>` is using `Struct`.
  * 
  * The properties and methods on the provable type exist in all base SnarkyJS types as well (aka. {@link Field}, {@link Bool}, etc.). In most cases, a zkApp developer does not need these functions to create Dapps.
  */
 declare interface Provable<T> {
-  toFields: (x: T) => Field[];
-  toAuxiliary: (x?: T) => any[];
-  fromFields: (x: Field[], aux: any[]) => T;
+  /**
+   * A function that takes `value`, an element of type `T`, as argument and returns an array of {@link Field} elements that made up the provable data of `value`.
+   * 
+   * @param value - the element of type `T` to generate the {@link Field} array from.
+   * 
+   * @return A {@link Field} array describing how this `T` element is made up of {@link Field} elements.
+   */
+  toFields: (value: T) => Field[];
+
+   /**
+   * A function that takes `value` (optional), an element of type `T`, as argument and returns an array of any type that made up the "auxilary" (non-provable) data of `value`.
+   * 
+   * @param value - the element of type `T` to generate the auxilary data array from, optional. If not provided, an empty array is returned.
```

Comment:
> If not provided, an empty array is returned.

That's actually not what it does!

```suggestion
   * @param value - the element of type `T` to generate the auxiliary data array from, optional. If not provided, a default value for auxiliary data is returned.
```